Packs/GoogleApigee/Integrations/GoogleApigeeEventCollector/GoogleApigeeEventCollector.py

```diff
@@ -0,0 +1,310 @@
+
+''' CLIENT CLASS '''
+
+
+class Client(BaseClient):
+    """Client class to interact with the service API
+
+    This Client implements API calls to the Saas Security platform, and does not contain any XSOAR logic.
+    Handles the token retrieval.
+
+    :param base_url (str): Saas Security server url.
+    :param username (str): Username.
+    :param password (str): Password.
+    :param verify (bool): specifies whether to verify the SSL certificate or not.
+    :param proxy (bool): specifies if to use XSOAR proxy settings.
+    :param org_name (str): the organization name
+    :param zone (str): the zone name
+    :param limit (int): the maximum logs to return per fetch call
+    """
+
+    def __init__(self, base_url: str, username: str, password: str, verify: bool, proxy: bool,
+                 org_name: str, zone: str, limit: int = DEFAULT_LIMIT, **kwargs):
+        self.username = username
+        self.password = password
+        self.org_name = org_name
+        self.max_fetch = limit
+        self.zone = zone
+
+        super().__init__(base_url=base_url, verify=verify, proxy=proxy, **kwargs)
+
+    def http_request(self, *args, **kwargs):
+        """
+        Overrides Base client request function, retrieves and adds to headers access token before sending the request.
+        """
+        token = self.get_access_token()
+        headers = {
+            'Authorization': f'Bearer {token}',
+            'Content-Type': 'application/json',
+        }
+        return super()._http_request(*args, headers=headers, **kwargs)  # type: ignore[misc]
+
+    def get_access_token(self) -> str:
+        """
+       Obtains access and refresh token from server.
+       Access token is used and stored in the integration context until expiration time.
+       After expiration, new access token are obtained and stored in the integration context.
+
+        Returns:
+            str: the access token.
+       """
+        integration_context = get_integration_context()
+        access_token = integration_context.get('access_token')
+        token_initiate_time = integration_context.get('token_initiate_time')
+        token_expiration_seconds = integration_context.get('token_expiration_seconds')
+        refresh_token = integration_context.get('refresh_token', '')
+
+        if access_token and Client.is_token_valid(
+            token_initiate_time=float(token_initiate_time),
+            token_expiration_seconds=float(token_expiration_seconds)
+        ):
+            return access_token
+        # There's no token or it is expired
+        access_token, token_expiration_seconds, refresh_token = self.get_token_request(refresh_token)
+        integration_context = {
+            'access_token': access_token,
+            'token_expiration_seconds': token_expiration_seconds,
+            'token_initiate_time': time.time(),
+            'refresh_token': refresh_token,
+        }
+        demisto.info('successfully updated access token')
```

Comment:
```suggestion
        demisto.debug('successfully updated access token')
```
Move this log to be after you save it to context (so if the saving won't complete, we won't think that it did)